Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -94,19 +95,48 @@ public function __construct(
      */
     public function handle(CommandInterface $command): void
     {
-        // the commands only calculate which events they want to have published, but do not do the
-        // publishing themselves
-        $eventsToPublishOrGenerator = $this->commandBus->handle($command);
-
-        if ($eventsToPublishOrGenerator instanceof EventsToPublish) {
-            $eventsToPublish = $this->enrichEventsToPublishWithMetadata($eventsToPublishOrGenerator);
-            $this->eventPersister->publishEvents($this, $eventsToPublish);
-        } else {
-            foreach ($eventsToPublishOrGenerator as $eventsToPublish) {
-                assert($eventsToPublish instanceof EventsToPublish); // just for the ide
-                $eventsToPublish = $this->enrichEventsToPublishWithMetadata($eventsToPublish);
-                $this->eventPersister->publishEvents($this, $eventsToPublish);
+        $toPublish = $this->commandBus->handle($command);
+
+        if ($toPublish instanceof EventsToPublish) {
+            // simple case
+            $eventsToPublish = $this->enrichEventsToPublishWithMetadata($toPublish);
+            if ($eventsToPublish->events->isEmpty()) {
+                return;
+            }
+            $this->eventPersister->publishWithoutKetchup($eventsToPublish);
```

Comment:
Did you really call it Ketchup to troll us? 🙃

Reply:
Maybe for jon :D 
It will be removed either way with the subscription engine:

> TODO Will be refactored via https://github.com/neos/neos-development-collection/pull/5321

:D

Reply:
adjusted as requested :)